appdo/appdo.py

```diff
@@ -14,194 +13,160 @@
-        '''
-        Merge config.
+        """Merge config.
+
         TODO: use better merging
-        '''
+        """
         self.config.update(conf)
 
     @staticmethod
-    def build_source_commands(conf):
-        '''
-        private method
-        '''
-        try:
-            srcfile = conf['source']
-            if isinstance(srcfile, str):
-                return ['source ' + srcfile]
-            else:
-                return ['source ' + x for x in srcfile]
-        except KeyError:
-            return []
+    def _build_source_commands(conf):
+        """private method."""
+        srcfile = conf.get('source')
+        if not srcfile:
+            return srcfile
+        if isinstance(srcfile, str):
+            return ['source {0}'.format(srcfile)]
+        else:
+            return ['source {0}'.format(x) for x in srcfile]
 
     @staticmethod
-    def build_cd_command(conf):
-        '''
-        private method
-        '''
-        try:
-            cmd = conf['cd']
-            if isinstance(cmd, str):
-                return ['cd ' + cmd]
-            else:
-                return []
-        except KeyError:
+    def _build_cd_command(conf):
+        """private method."""
+        cmd = conf['cd']
+        if not cmd:
+            return cmd
+        if isinstance(cmd, str):
+            return ['cd {0}'.format(cmd)]
+        else:
             return []
 
     @staticmethod
-    def build_before_commands(conf):
-        '''
-        private method
-        '''
-        try:
-            cmd = conf['before']
-            if isinstance(cmd, str):
-                return [cmd]
-            else:
-                return cmd
-        except KeyError:
-            return []
+    def _build_before_commands(conf):
+        """private method."""
+        cmd = conf['before']
+        if not cmd:
+            return cmd
+        if isinstance(cmd, str):
+            return [cmd]
+        else:
+            return cmd
 
     @staticmethod
-    def build_envs(conf):
-        '''
-        private method
-        '''
-        try:
-            envs = conf['env']
-            if isinstance(envs, dict):
-                return [k + '=' + v for k,v in envs.items()]
-            else:
-                return []
-        except KeyError:
+    def _build_envs(conf):
+        """private method."""
+        envs = conf['env']
+        if not envs:
+            return envs
+        if isinstance(envs, dict):
+            return ['{0}={1}'.format(k, v) for k, v in envs.items()]
+        else:
             return []
 
     @staticmethod
-    def build_prefix_command(conf):
-        '''
-        private method
-        '''
-        try:
-            cmd = conf['prefix']
-            if isinstance(cmd, str):
-                return [cmd + ' ']
-            else:
-                return []
-        except KeyError:
+    def _build_prefix_command(conf):
+        """private method."""
+        cmd = conf['prefix']
```

Comment:
and `conf.get('prefix')`.

I should have written this in higher-order function manner...